pyproject.toml

```diff
@@ -0,0 +1,138 @@
+django-reversion = "~3.0"           # Model version history with middleware hooks to all post_save signals
+django-guardian = "~2.2"            # Per Object permissions framework
+django-taggit = "~1.2"              # Generic multi-model tagging library
+django-taggit-serializer = "~0.1"   # REST Framework serializers for Django-taggit
+APScheduler = "~3.6"                # Scheduler
+redwine = { git = "https://github.com/dotkom/redwine.git", tag = "v1.3.1" }
+reportlab = "~3.5"
+pdfdocument = "~3.3"
+Unidecode = "~1.1"                  # Translates every unicode symbol to the closest ascii. online_mail_usernames
+django-markdown-deux = "~1.0"
+django-cors-headers = "~3.2"        # Enable CORS for Nibble
+django-datetime-widget = { git = "https://github.com/oleast/django-datetime-widget.git", rev = "95a5ba4738853d1ed4081e1e33e1600cba984aad" }
+django-webpack-loader = "~0.6"      # Integration with webpack
+dj-database-url = "~0.5"            # Allows to configure databases using URLs
+python-decouple = "~3.1"            # Configuration and settings management
+django-oidc-provider = "~0.7"       # OpenID Connect Provider
+docutils = "~0.15"                  # Allows django-admindocs to generate documentation
+requests-oauthlib = "~1.2"          # Simplify OAuth2 connections to external apps
+oic = "~1.0"			            # OpenID Connect Client
+python-dateutil = "~2.8"            # Simplify date parsing
+coreapi = "~2.3"                    # API Documentation
+requests = { version = "~2.22", extras = [ "security" ] }
+# Wiki
+wiki = "0.6b1"
+
+# Django 1.9 support
+sorl-thumbnail = "~12.6"
+django-appconf = "~1.0"
+
+# non-pip
+# imagemagick                   # From package repository. Tested with version == 8:6.7.7.10.
+# libjpeg-dev					# From package repository.
+# libpq-dev						# From package repository.
+
+# Monitoring
+raven = "~6.10"
+
+django-js-reverse = "~0.9"
+
+psycopg2-binary = { version = "~2.8", optional = true }
+uwsgi = { version = "~2.0", optional = true }
+redis = { version = "~3.3", optional = true }
+
+# Docs
+sphinx = { version = "*", optional = true }
+sphinx-rtd-theme = { version = "*", optional = true }
+
+
+[tool.poetry.dev-dependencies]
+# Testing
+pytest = "~5.2"
+pytest-cov = "~2.8"
+pytest-django = "~3.6"
+factory-boy = "~2.11"
+freezegun = "~0.3"
+mock = "~3.0"
+tox = "~3.14"
+
+# Linting
+flake8 = "~3.7"
+isort = "~4.3"
+pycodestyle = "~2.5"
+black = "19.10b0"
+
+[tool.poetry.extras]
+prod = ["psycopg2-binary", "uwsgi", "redis"]
+docs = ["sphinx", "sphinx-rtd-theme"]
+
+[tool.tox]
+legacy_tox_ini = """
+[tox]
+envlist = tests, flake8, isort, black
+skipsdist = True
+
+[testenv]
+basepython = python3
+whitelist_externals =
+    npm
+    py.test
+    isort
+    flake8
+    black
+commands =
+    isort: isort -rc -c apps middleware scripts utils
+    flake8: flake8 apps middleware scripts utils
+    black: black --check apps middleware scripts utils onlineweb4
```

Comment:
For some reason, Drone is unable to find these "external commands", but Github Actions can find them just fine?

Reply:
This turned out to be quite the complicated task, as Drone doesn't cache _everything_ between steps, only the workspace, so links in `usr/local/bin`, and the downloaded pip-packages are not kept. I solved this by installing during each step, and setting up so that the pip-cache is kept between steps.